exercises/error-handling/.meta/src/reference/java/ErrorHandling.java

```diff
@@ -0,0 +1,38 @@
+import java.util.Optional;
+
+class ErrorHandling {
+
+    void handleErrorByThrowingException() throws Exception {
```

Comment:
In general it's considered bad practice to throw or catch just `Exception`. You should either use one of the built in [unchecked exceptions](https://www.geeksforgeeks.org/checked-vs-unchecked-exceptions-in-java/) (e.g. `IllegalArgumentException`) or you should construct your own checked exception which extends `Exception`. This has been done for other exercises which don't just focus on error handling so I definitely don't think it's too much to ask in this exercise. On the contrary, I think it's an important thing to learn about error handling in Java 🙂 Since you've estimated this exercise as difficulty 4, you would need to provide starter implementation for the custom exception, so I don't think it should be that hard for the user to implement 🙂 What do you think? 🙂

Reply:
Good point 🙂I will add a test case for throwing a custom exception and provide starter implementation 🙂

Reply:
Great, thank you! Those new methods look great! 🙂 

However, I would still disagree with keeping these methods which throw `Exception`. I realise I didn't explain why it's bad practice to throw or catch `Exception`, sorry about that! The reason is that `Exception` is the superclass of all exceptions. So if you say your method throws `Exception` then you don't give any details about the exception your method might throw, it could throw any exception at all. Even worse, when you catch `Exception` you catch all exceptions including runtime exceptions you might not want to handle. Therefore in Java you should always throw and catch specific exception classes that accurately describe the error that has occurred or the error that you're trying to handle. [This stackoverflow question](https://stackoverflow.com/questions/2416316/why-is-the-catchexception-almost-always-a-bad-idea) has some more information. Does that make sense? 🙂 

It might actually be a good idea to add some hints to this exercise (in a hints.md file in the .meta directory, see [flatten-array](https://github.com/exercism/java/blob/master/exercises/flatten-array/.meta/hints.md) for an example) explaining the difference between checked and unchecked exceptions (possibly linking to https://www.geeksforgeeks.org/checked-vs-unchecked-exceptions-in-java/), when you should use unchecked exceptions (possibly linking to https://docs.oracle.com/javase/tutorial/essential/exceptions/runtime.html) and why you shouldn't throw or catch `Exception` (possibly linking to the stackoverflow question). Do you want to do that @CRivasGomez? That's absolutely not a requirement to get this exercise merged, if you think you have enough work with the rest of the exercise then I can just open a separate issue for adding a hints file 🙂